### PR TITLE
ROR capstone project

### PR DESCRIPTION
**Pull request by Hadi111jafari**
PR Description:
in this PR I implemented:

Splash screen
1- A simple page with the name of the Budget App and links to the sign-up and log-in pages.

Sign up and log in to pages
1- The user is able to register in the app with full name (first name and Lastname), email, and password (all mandatory).
2- The user can log into the app using email and password.
3- If the user is not logged in, they can't access pages that require the user to be logged in.

Category page
1- When the user logs in, they are presented with the categories page.
2- For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
3- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
4- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

Transactions page
1- For a given category, the list of transactions is presented, ordered by the most recent.
2- At the top of the page the user can see the total amount for the category.
3- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
4- When the user clicks on the "Back" button (<), the user navigates to the home page.

Add a new category" page
1- The user fills out a form to create a new category, indicating their name and icon.
2- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
3- When the user clicks on the "Back" button (<), the user navigates to the home page.

Add a new transaction" page
1- The user fills out a form to create a new transaction with:
2- name
3- amount
4- The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
5- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

Testing requirements
1- Create unit and integration tests for all the most important components of your RoR application.

Technical requirements
1- Used Postgres as a database.
2- Used devise for authentication.
3- validate all user input to make sure that anyone with bad intentions cannot compromise your app.
4- The project is deployed and accessible online.
5- database schema reflects the requirements structure

## Vedio Presentation
[click here](https://www.loom.com/share/9212b70a80dc49d1bcd4b15a999f9108)

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (andersonlebon) [inline app/views/devise/registrations/new.html.erb]: - [x] Kindly make sure the user name is provided when the user is signing up as the name attribute should be present in your validation.

- [x] Kindly also follow the template design on your both registration page and login pages

image

File: app/views/groups/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      <span>
+        $ <%= expense.amount %>
+      </span>
+    </div>     
+    <% end %>
+    
+</section>
+<button class="btn">
+    <%= link_to "add a new transaction", "/expenses/new" %>
+</button>
```
Comment (andersonlebon) [inline app/views/groups/show.html.erb]: - [x] Kindly display the list of all expenses related to this category/group
- [x] Also make the list follow the template design

 | the template | Your design|
--|--
image | image

File: app/views/groups/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    <div class="name-icon">
+      <span>
+        <%= group.icon %>
+      </span> 
+      <h4>
+        <%= group.name %>
+      <h4>
+    </div>  
+      <p>
+        $ <%= group.total_amount %>
```
Comment (andersonlebon) [inline app/views/groups/index.html.erb]: - [x] Kindly make sure the total of all expenses related to each category/group is displayed here.

File: app/views/groups/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  <div class="select_icon">
+    <%= form.label :icon, 'Select Icon:', class: "select_icon_label" %>
+    <%= form.select :icon, ["✈️", "🎮", "📚", "🏥", "🏈", "🎬", "🏠", "🐶", "🍻", "🍔", "🚗", "🖥"], value: @group.icon, required: true, class: "select_icon_menu" %>
+  </div>
+
+  <div class="actions">
+    <%= form.submit "Save", class: "app_button green_button" %>
+  </div>
+<% end %>
+</section>
```
Comment (andersonlebon) [inline app/views/groups/new.html.erb]: - [x] Kindly make sure this form looks like the original template(navigation, buttons, inputs ,,,)
- [x] Also use icons in your navigation bar
you design :
image

File: app/views/expenses/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  <div class="field">
+    <%= form.text_field :amount, value: @expense.amount, placeholder: "Amount", type: "float", required: true, class: "form_field" %>
+  </div>
+
+  <div class="actions_expense">
+    <%= form.submit "Save", class: "app_button green_button" %>
+  </div>
+<% end %>
+</section>
+
```
Comment (andersonlebon) [inline app/views/expenses/new.html.erb]: - [x] Kindly make sure this form looks like the original template(navigation, buttons, inputs ,,,)
- [x]  Also use icons in your navigation bar

you design :

image

File: app/views/devise/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (andersonlebon) [inline app/views/devise/sessions/new.html.erb]: - [x] Kindly follow the design template here on the login page
image

File: README.md [hunk 7]
```diff
@@ -1,24 +1,66 @@
...
+Set up database direct them to the config/database.yml file
+```
+bundle install
+```
+```
+'npm install' or 'yarn install'
+```
+```
+rails db:create
+```
```
Comment (andersonlebon) [inline README.md]: - [x] Kindly follow the documentation requirement and provide a video demo of your project, it should match these criterias:

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Hadi111jafari 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Hadi111jafari** [comment]
The error from deployment is from Netlify that failed and I deployed in heroku

**Review by AshakaE — APPROVED** [review]
Hi @Hadi111jafari , 

Thanks for making the requested changes 👏 

### Highlights ✨ 

- Pull request 👌 
- Git flow ✅ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**